event_files/topdown.json

```diff
@@ -0,0 +1,96 @@
+[
+    {
+        "EventName": "TOPDOWN.SLOTS",
+        "EventCode": "0x0",
+        "BriefDescription": "TMA slots available for an unhalted hart",
+        "PublicDescription": "TMA slots available for an unhalted hart"
+    },
+    {
+        "MetricName": "TOPDOWN.SLOTS_V2",
+        "BriefDescription": "TMA slots available for an unhalted hart",
+        "PublicDescription": "TMA slots available for an unhalted hart",
+        "MetricExpr": "pipeline_width * GEN.CYCLES.SMT.CORE"
+    },
+    {
+        "EventName": "TOPDOWN.FRONTEND_BOUND.SLOTS",
+        "EventCode": "0x0",
+        "BriefDescription": "TMA slots unused due to the frontend did not supply enough operations",
+        "PublicDescription": "TMA slots unused due to the frontend did not supply enough operations"
+    },
+    {
+        "EventName": "TOPDOWN.BAD_SPECULATION.SLOTS",
+        "EventCode": "0x0",
+        "BriefDescription": "TMA slots wasted due to incorrect speculations.",
+        "PublicDescription": "TMA slots wasted due to incorrect speculations. This include slots used to issue uops that do not eventually get retired and slots for which the issue-pipeline was blocked due to recovery from earlier incorrect speculation. For example; wasted slots due to miss-predicted branches should be accounted by this event. Incorrect data speculation followed by Memory Ordering Nukes is another example."
+    },
+    {
+        "MetricName": "TOPDOWN.BAD_SPECULATION.SLOTS_V2",
+        "BriefDescription": "TMA slots wasted due to incorrect speculations",
+        "MetricExpr": "SPEC.UOP_ISSUED - RETIRED.UOP + pipeline_width * PRD.PIPELINE_FLUSH.RECOVERY_CYCLES"
+    },
+    {
+        "EventName": "TOPDOWN.BAD_SPECULATION.CONTROL_FLOW.SLOTS",
+        "EventCode": "0x0",
+        "BriefDescription": "TMA slots wasted due to incorrect control flow speculations.",
+        "PublicDescription": "TMA slots wasted due to incorrect control flow speculations."
+    },
+    {
+        "EventName": "TOPDOWN.BAD_SPECULATION.MEM_ORDERING.SLOTS",
+        "EventCode": "0x0",
+        "BriefDescription": "TMA slots wasted due to memory ordering violations.",
+        "PublicDescription": "TMA slots wasted due to memory ordering violations."
+    },
```

Comment:
Of course there are many other sources of bad spec, including interrupts and exceptions.  I don't know what we need events for those, I'm more curious how we talk about summing sub-events up to the parent events.  Will we say that there is always an implicit "other" category, and hence sub-events never sum up to the parent?  Is that true for all L1 parent events?

Reply:
The first 1 or 2 TMA levels should sum up I beleive. So in this particular case it makes sense to add explicit  TOPDOWN.BAD_SPECULATION.ORDER metric with formala TOPDOWN.BAD_SPECULATION - (TOPDOWN.BAD_SPECULATION.CONTROL_FLOW + TOPDOWN.BAD_SPECULATION.MEM_ORDERING). Will do this.